src/operators/InterestDistributor.sol

```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.26;
+
+import {IERC20} from "src/interfaces/IERC20.sol";
+import {IERC7540Vault} from "src/interfaces/IERC7540.sol";
+import {IPoolManager} from "src/interfaces/IPoolManager.sol";
+import {MathLib} from "src/libraries/MathLib.sol";
+import {IInterestDistributor, InterestDetails} from "src/interfaces/operators/IInterestDistributor.sol";
+
+/// @title  InterestDistributor
+/// @notice Contract that can be enabled by calling vault.setOperator(address(interestDistributor), true), which then
+///         allows permissionless triggers of redeem requests for the accrued interest on a vault.
+///
+///         Whenever a user claims new tranche tokens or a principal redeem request is initiated,
+///         distribute() should be called to update the outstanding shares that the interest is computed based on.
+///
+///         Uses peak price to calculate interest. If the price goes down, interest will only be redeemed
+///         again once the price fully recovers above the previous high point (peak).
+///         Peak is stored per user since if the price has globally gone down, and a user invests at that time,
+///         they'd expect to redeem interest based on the price they invested in, not the previous high point.
+/// @dev    Requires that:
+///         - Whenever orders are fulfilled, an UpdateTranchePrice message is also submitted
+///         - Users claim the complete fulfilled amount using the router, rather than partial claims through the vault.
+///         Otherwise the interest amounts may be off.
+contract InterestDistributor is IInterestDistributor {
+    using MathLib for uint256;
+
+    mapping(address vault => mapping(address user => InterestDetails)) internal _users;
+
+    IPoolManager public immutable poolManager;
+
+    constructor(address poolManager_) {
+        poolManager = IPoolManager(poolManager_);
+    }
+
+    /// @inheritdoc IInterestDistributor
+    function distribute(address vault, address controller) external {
+        IERC7540Vault vault_ = IERC7540Vault(vault);
+        require(vault_.isOperator(controller, address(this)), "InterestDistributor/not-an-operator");
+
+        InterestDetails storage user = _users[vault][controller];
```

Comment:
```suggestion
        InterestDetails memory user = _users[vault][controller];
```

Reply:
Even though this is a single slot, each `user.*` down the code costs 100 gas.

Reply:
True